packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -821,7 +829,7 @@ class TextPainter {
       final TextBox box = boxes.last;
       final double caretStart = box.start;
       final double dx = box.direction == TextDirection.rtl ? caretStart - caretPrototype.width : caretStart;
-      return Rect.fromLTRB(min(dx, _paragraph!.width), box.top, min(dx, _paragraph!.width), box.bottom);
+      return Rect.fromLTRB(dx.clamp(0, _paragraph!.width), box.top, dx.clamp(0, _paragraph!.width), box.bottom);
```

Comment:
So I fixed it by adding clamping in method
https://github.com/flutter/flutter/blob/04156b66364b6be646432cc77487c108fc19f378/packages/flutter/lib/src/rendering/editable.dart#L1519
I think it should be limited similar to caret metrics computation.

Reply:
So I fixed it by adding clamping in method
https://github.com/flutter/flutter/blob/04156b66364b6be646432cc77487c108fc19f378/packages/flutter/lib/src/rendering/editable.dart#L1519
I think it should be constrained, like in caret metrics computation (because both methods use `getRectsForRange`)

Reply:
So I fixed it by adding clamping in method
https://github.com/flutter/flutter/blob/04156b66364b6be646432cc77487c108fc19f378/packages/flutter/lib/src/rendering/editable.dart#L1519
I think x coordinate in the result should be clamped as when compute caret metrics (because both methods use `getRectsForRange`)